src/Microsoft.DotNet.Build.Tasks/PackageFiles/buildagainstpackages.targets

```diff
@@ -38,11 +38,11 @@
     </ResolveNuGetPackagesDependsOn>
   </PropertyGroup>
 
-  <Target Name="GenerateTestProjectJson" DependsOnTargets="ExtractProjectReferences;GatherProjectReferenceForProjectJson;DetermineProjectJsonPath;AddDependenciesToProjectJson" />
+  <Target Name="GenerateTestProjectJson" DependsOnTargets="ExtractProjectReferences;GatherProjectReferenceForProjectJson;DetermineProjectJsonPath;AddDependenciesToProjectJson"/>
 
   <Target Name="GenerateAllTestProjectJsons" DependsOnTargets="GenerateTestProjectJson" />
 
-  <Target Name="DetermineProjectJsonPath">
+  <Target Name="DetermineProjectJsonPath" Condition="'$(UsesCommonTestProjectJson)'!='true'">
```

Comment:
Made the associated change. Since the propertygroup listed above is global, I had to create another target to evaluate in the given context.